tests/test_index.c

```diff
@@ -12,564 +16,1028 @@
+
+	// Index should now have CRC64 and SHA256
+	assert_uint_eq(lzma_index_checks(index),
+			LZMA_INDEX_CHECK_MASK_CRC64 |
+			LZMA_INDEX_CHECK_MASK_SHA256);
+
+	lzma_index_end(index, NULL);
+}
 
-		if (ra.stream.block_count == 0)
-			continue;
 
-		if (ra.block.number_in_file != rb.block.number_in_file
-				|| ra.block.compressed_file_offset
-					!= rb.block.compressed_file_offset
-				|| ra.block.uncompressed_file_offset
-					!= rb.block.uncompressed_file_offset
-				|| ra.block.number_in_stream
-					!= rb.block.number_in_stream
-				|| ra.block.compressed_stream_offset
-					!= rb.block.compressed_stream_offset
-				|| ra.block.uncompressed_stream_offset
-					!= rb.block.uncompressed_stream_offset
-				|| ra.block.uncompressed_size
-					!= rb.block.uncompressed_size
-				|| ra.block.unpadded_size
-					!= rb.block.unpadded_size
-				|| ra.block.total_size
-					!= rb.block.total_size)
-			return false;
-	}
+static void
+test_lzma_index_stream_padding(void)
+{
+	// Test NULL index
+	assert_lzma_ret(lzma_index_stream_padding(NULL, 0),
+			LZMA_PROG_ERROR);
+
+	lzma_index *index = lzma_index_init(NULL);
+	assert_true(index != NULL);
+
+	// Test stream padding not a multiple of 4
+	assert_lzma_ret(lzma_index_stream_padding(index, 3),
+			LZMA_PROG_ERROR);
+
+	// Test stream padding too large
+	assert_lzma_ret(lzma_index_stream_padding(index, LZMA_VLI_MAX + 1),
+			LZMA_PROG_ERROR);
+
+	// Test stream padding valid
+	assert_lzma_ret(lzma_index_stream_padding(index, 0x1000),
+			LZMA_OK);
+	assert_lzma_ret(lzma_index_stream_padding(index, 4),
+			LZMA_OK);
+	assert_lzma_ret(lzma_index_stream_padding(index, 0),
+			LZMA_OK);
+
+	// Test stream padding causing the file size to grow too large
+	assert_lzma_ret(lzma_index_append(index, NULL,
+			LZMA_VLI_MAX - 0x1000, 1), LZMA_OK);
+	assert_lzma_ret(lzma_index_stream_padding(index, 0x1000),
+			LZMA_DATA_ERROR);
+
+	lzma_index_end(index, NULL);
 }
 
 
 static void
-test_equal(void)
+test_lzma_index_stream_count(void)
 {
-	lzma_index *a = create_empty();
-	lzma_index *b = create_small();
-	lzma_index *c = create_big();
-	expect(a && b && c);
-
-	expect(is_equal(a, a));
-	expect(is_equal(b, b));
-	expect(is_equal(c, c));
-
-	expect(!is_equal(a, b));
-	expect(!is_equal(a, c));
-	expect(!is_equal(b, c));
-
-	lzma_index_end(a, NULL);
-	lzma_index_end(b, NULL);
-	lzma_index_end(c, NULL);
+	lzma_index *index = lzma_index_init(NULL);
+	assert_true(index != NULL);
+
+	assert_uint_eq(lzma_index_stream_count(index), 1);
+
+	// Appending blocks should not change the stream count value
+	assert_lzma_ret(lzma_index_append(index, NULL, UNPADDED_SIZE_MIN,
+			1), LZMA_OK);
+
+	assert_uint_eq(lzma_index_stream_count(index), 1);
+
+	// Test with multiple streams
+	for (int i = 0; i < 100; i++) {
+		lzma_index *idx = lzma_index_init(NULL);
+		assert_true(idx != NULL);
+		assert_lzma_ret(lzma_index_cat(index, idx, NULL), LZMA_OK);
+		assert_uint_eq(lzma_index_stream_count(index), i + 2);
```

Comment:
This triggers -Wsign-conversion. Here it doesn't matter if i is signed or unsigned so perhaps unsigned int would be better. (If i starts 2 to then i + 2 isn't needed, both are fine.)